utils/polus-bfio-util/bfio/bfio.py

```diff
@@ -55,194 +82,356 @@ class BioReader():
-    
-    ''' Get raw metadata '''
-    def read_metadata(self,update=False):
+        For information on the OMEXML class:
+        https://github.com/CellProfiler/python-bioformats/blob/master/bioformats/omexml.py
+
+        Args:
+            update (bool, optional): Whether to force update of metadata. Defaults to False.
+                Only needs to be used if the metadata is updated while the image is open.
+
+        Returns:
+            OMEXML: Class that 
+        """
+        
+        # Return cached data if it exists
         if self._metadata and not update:
             return self._metadata
+        
         # For some reason, tif files need to use the generic ImageReader while everything else
         # can use the OMETiffReader.
         if self._file_path.endswith('.ome.tif'):
             rdr = jutil.JClassWrapper('loci.formats.in.OMETiffReader')()
         else:
             rdr = jutil.JClassWrapper('loci.formats.ImageReader')()
         rdr.setOriginalMetadataPopulated(True)
-        clsOMEXMLService = jutil.JClassWrapper('loci.formats.services.OMEXMLService')
-        serviceFactory = jutil.JClassWrapper('loci.common.services.ServiceFactory')()
+        
+        # Access the OMEXML Service
+        clsOMEXMLService = jutil.JClassWrapper(
+            'loci.formats.services.OMEXMLService')
+        serviceFactory = jutil.JClassWrapper(
+            'loci.common.services.ServiceFactory')()
         service = serviceFactory.getInstance(clsOMEXMLService.klass)
         omexml = service.createOMEXMLMetadata()
+        
+        # Read the metadata
         rdr.setMetadataStore(omexml)
         rdr.setId(self._file_path)
-        
+
+        # Parse it using the OMEXML class
         self._metadata = bioformats.OMEXML(omexml.dumpXML())
         return self._metadata
-    
-    ''' Utility functions to validate image dimensions '''
-    def _val_xyz(self,xyz,axis):
+
+    def _val_xyz(self, xyz, axis):
+        """_val_xyz Utility function for validating image dimensions
+
+        Args:
+            xyz (int): Pixel value of x, y, or z dimension.
+                If None, returns the maximum range of the dimension
+            axis (str): Must be 'x', 'y', or 'z'
+
+        Returns:
+            list: list of ints indicating the first and last index in the dimension
+        """
+        assert axis in 'XYZ'
         if not xyz:
-            xyz = [0,self._xyzct[axis]]
+            xyz = [0, self._xyzct[axis]]
         else:
-            if len(xyz)!=2:
-                ValueError('{} must be a list or tuple of length 2.'.format(axis))
-            elif xyz[0]<0:
-                ValueError('{}[0] must be greater than or equal to 0.'.format(axis))
-            elif xyz[1]>self._xyzct[axis]:
-                ValueError('{}[1] cannot be greater than the maximum of the dimension ({}).'.format(axis,self._xyzct[axis]))
+            assert len(xyz) == 2,\
+                '{} must be a list or tuple of length 2.'.format(axis)
```

Comment:
why is this true? Couldn't it be one? just a long strip of pixels?

Reply:
It just provides a range of values (min/max). If you wanted to get just a strip of pixels, you would do something like [1,2], which would get a strip of pixels for index 1.